src/Psalm/Config.php

```diff
@@ -955,6 +984,40 @@ private static function fromXmlAndPaths(
             $base_dir = $current_dir;
         }
 
+        $composer_json_path = Composer::getJsonFilePath($config->base_dir);
+
+        $composer_json = null;
+        if (file_exists($composer_json_path)) {
+            if (!$composer_json = json_decode(file_get_contents($composer_json_path), true)) {
+                throw new UnexpectedValueException('Invalid composer.json at ' . $composer_json_path);
```

Comment:
Should we revert to runtime extension if we failed to parse composer file?

Reply:
We do the same thing when getting the PHP version from composer.json.